core/src/provider/rpc.rs

```diff
@@ -19,13 +19,28 @@ pub struct RpcBlockDataProvider {
     pub provider: ReqwestProvider,
     pub client: RpcClient<Http<Client>>,
     block_number: u64,
+
+    #[cfg(any(test, feature = "test-utils"))]
+    pub persistent_block_data:
+        std::sync::Arc<tokio::sync::Mutex<super::persistent_map::PersistentBlockData>>,
 }
 
 impl RpcBlockDataProvider {
     pub fn new(url: &str, block_number: u64) -> RaikoResult<Self> {
         let url =
             reqwest::Url::parse(url).map_err(|_| RaikoError::RPC("Invalid RPC URL".to_owned()))?;
+
         Ok(Self {
+            #[cfg(any(test, feature = "test-utils"))]
```

Comment:
it’s better to avoid excessive conditional compilation test code in the main program logic. I suggest making another RpcBlockDataProvider trait impl for test.

Reply:
Good call. Creating another RPC provider might be better. Thanks will update soon